### PR TITLE
Document HMatrix classes

### PR DESCRIPTION
**Pull request by jschueller**

### CODE REVIEW

File: python/src/HMatrixFactory_doc.i.in [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+
+%feature("docstring") OT::HMatrixFactory::build
+"Build a compressed matrix.
+
+Parameters
+----------
+sample : 2-d sequence of float
+    Matrix data.
+outputDimension : int
+    ??
```
Comment (regislebrun) [inline python/src/HMatrixFactory_doc.i.in]: !!

Comment (jschueller) [inline python/src/HMatrixFactory_doc.i.in]: I did not fully understand what this parameter stands for.

Comment (jschueller) [inline python/src/HMatrixFactory_doc.i.in]: @dbarbier by the way, should ot rely on hmat >= 1.2 now ?

**Comment by dbarbier** [comment]
I pushed changes into my repo, hopefully it is clearer now.

**Comment by dbarbier** [comment]
Err, sorry, there were docstring errors, this is fixed in my repo.

**Comment by jschueller** [comment]
ok, I squashed thme here and fixed some warnings in your lib changes.

**Comment by dbarbier** [comment]
I just pushed minor changes, feel free to take this commit if other changes are needed.

**Comment by jschueller** [comment]
Ok, I squashed them.

**Pull request merged**